### PR TITLE
Fix ids api

### PR DESCRIPTION
**Pull request by microwavedcola1**
Signed-off-by: microwavedcola1 <microwavedcola@gmail.com>

### CODE REVIEW

File: ts/client/src/accounts/group.ts [hunk 1]
```diff
@@ -308,7 +310,7 @@ export class Group {
       await client.program.provider.connection.getMultipleAccountsInfo(oracles);
 
     const coder = new BorshAccountsCoder(client.program.idl);
-    ais.forEach(async (ai, i) => {
+    for await (const [i, ai] of ais.entries()) {
```
Comment (microwavedcola1) [inline ts/client/src/accounts/group.ts:313]: the await here is key

File: ts/client/src/accounts/group.ts [hunk 2]
```diff
@@ -90,7 +90,9 @@ export class Group {
     public vaultAmountsMap: Map<string, BN>,
   ) {}
 
-  public async reloadAll(client: MangoClient, ids?: Id): Promise<void> {
+  public async reloadAll(client: MangoClient): Promise<void> {
```
Comment (microwavedcola1) [inline ts/client/src/accounts/group.ts:93]: removing the ids option, not all pathways were using ids json before, now once configured on the client during building the client, the gpa vs api is used implicitly

File: ts/client/src/accounts/group.ts (hunk 1 above)
Comment (tylersssss) [reply ts/client/src/accounts/group.ts:313]: We may be able to make this faster by using Promise.all

Comment (tylersssss) [reply ts/client/src/accounts/group.ts:313]: something like this:

```
const loadPrices = Array.from(ais.entries()).map(async ([i, ai]) => {
      const perpMarket = perpMarkets[i];
      if (!ai)
        throw new Error(
          `Undefined ai object in reloadPerpMarketOraclePrices for ${perpMarket.oracle}!`,
        );
      const { price, uiPrice } = await this.decodePriceFromOracleAi(
        coder,
        perpMarket.oracle,
        ai,
        perpMarket.baseDecimals,
        client,
      );
      perpMarket._price = price;
      perpMarket._uiPrice = uiPrice;
    });

    await Promise.all(loadPrices);
    ```

Comment (microwavedcola1) [reply ts/client/src/accounts/group.ts]: thanks, fixed

**Review by tylersssss — APPROVED** [review]

**Pull request merged**